File: Magick.NET/Drawables/DrawableColor.cpp
//=================================================================================================
// Copyright 2013-2014 Dirk Lemstra <https://magick.codeplex.com/>
//
// Licensed under the ImageMagick License (the "License"); you may not use this file except in 
// compliance with the License. You may obtain a copy of the License at
//
//   http://www.imagemagick.org/script/license.php
//
// Unless required by applicable law or agreed to in writing, software distributed under the
// License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
// express or implied. See the License for the specific language governing permissions and
// limitations under the License.
//=================================================================================================
#include "Stdafx.h"
#include "DrawableColor.h"

namespace ImageMagick
{
	//==============================================================================================
	DrawableColor::DrawableColor(double x, double y, ImageMagick::PaintMethod paintMethod)
	{
		BaseValue = new Magick::DrawableColor(x, y, (Magick::PaintMethod)paintMethod);
	}
	//==============================================================================================
	ImageMagick::PaintMethod DrawableColor::PaintMethod::get()
	{
		return (ImageMagick::PaintMethod)Value->paintMethod();
	}
	//==============================================================================================
	void DrawableColor::PaintMethod::set(ImageMagick::PaintMethod value)
	{
		Value->paintMethod((Magick::PaintMethod)value);
	}
	//==============================================================================================
	double DrawableColor::X::get()
	{
		return Value->x();
	}
	//==============================================================================================
	void DrawableColor::X::set(double value)
	{
		Value->x(value);
	}
	//==============================================================================================
	double DrawableColor::Y::get()
	{
		return Value->y();
	}
	//==============================================================================================
	void DrawableColor::Y::set(double value)
	{
		Value->y(value);
	}
	//==============================================================================================
}

File: src/Magick.NET/Shared/Drawables/DrawableColor.cs
// Copyright 2013-2019 Dirk Lemstra <https://github.com/dlemstra/Magick.NET/>
//
// Licensed under the ImageMagick License (the "License"); you may not use this file except in
// compliance with the License. You may obtain a copy of the License at
//
//   https://www.imagemagick.org/script/license.php
//
// Unless required by applicable law or agreed to in writing, software distributed under the
// License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
// either express or implied. See the License for the specific language governing permissions
// and limitations under the License.

namespace ImageMagick
{
    /// <summary>
    /// Draws color on image using the current fill color, starting at specified position, and using
    /// specified paint method.
    /// </summary>
    public sealed class DrawableColor : IDrawable, IDrawingWand
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="DrawableColor"/> class.
        /// </summary>
        /// <param name="x">The X coordinate.</param>
        /// <param name="y">The Y coordinate.</param>
        /// <param name="paintMethod">The paint method to use.</param>
        public DrawableColor(double x, double y, PaintMethod paintMethod)
        {
            X = x;
            Y = y;
            PaintMethod = paintMethod;
        }

        /// <summary>
        /// Gets or sets the PaintMethod to use.
        /// </summary>
        public PaintMethod PaintMethod { get; set; }

        /// <summary>
        /// Gets or sets the X coordinate.
        /// </summary>
        public double X { get; set; }

        /// <summary>
        /// Gets or sets the Y coordinate.
        /// </summary>
        public double Y { get; set; }

        /// <summary>
        /// Draws this instance with the drawing wand.
        /// </summary>
        /// <param name="wand">The want to draw on.</param>
        void IDrawingWand.Draw(DrawingWand wand) => wand?.Color(X, Y, PaintMethod);
    }
}